### DISCUSSION TITLE
segment anything

### DISCUSSION THREAD

**Discussion by scott-vsi** [kornia/kornia#2711] [Potential Issue] [answered]
Would you expect the output of the segment anything model (SAM) using the [VisualPrompter](https://kornia.readthedocs.io/en/latest/models/segment_anything.html#visual-prompter) to match the output from the original FAIR version? 

I am not getting the same thing.

**Comment by scott-vsi** [comment, marked as answer]
Never mind. I was loading the image myself, instead of using kornia.io.load_image, so it was not scaled to between 0 and 1. Perhaps a warning could be printed if the image is not in the expected range? At the very least, it would be nice if the [documentation](https://kornia.readthedocs.io/en/latest/models/segment_anything.html#visual-prompter)/[tutorial](https://kornia.github.io/tutorials/nbs/image_prompter.html) mentioned this.

**Reply by johnnv1** [reply]
I think it makes sense to have a warning if it's not in range and comment on the documentation. Could you contribute a PR with these changes?

**Reply by scott-vsi** [reply]
Sure. Which is preferred, the ImagePrompter or the VisualPrompter

**Reply by edgarriba** [reply]
ImagePrompter is an alias that should be removed sooner or later. I’m more in favour of updating properly documentation rather than runtime checking which means adding extra O(n) tensor traverse. General rule in Kornia is operating floating point tensors in range 0/1

**Comment by scott-vsi** [comment]
Just circling back. I noticed that this is already mentioned in the documentation for [VisualPrompter .set_image](https://github.com/kornia/kornia/blob/v0.7.1/kornia/contrib/visual_prompter.py#L109).

I have created a PR to update the kornia documentation: https://github.com/kornia/kornia/pull/2738
My PR to update the kornia tutorial was merged: https://github.com/kornia/tutorials/pull/76
